Sources/AppTeamToolBox/MNQueueIterator.swift

```diff
@@ -43,12 +43,66 @@ public struct MNQueueIterator<Element: Identifiable> {
         fixIndexIntoRange()
     }
 
+    public mutating func insertAt(_ element: Element, index: Int) {
+        if index >= count {
+            items.append(element)
+        } else {
+            items.insert(element, at: index)
+        }
+        fixIndexIntoRange()
+    }
+
     public mutating func insertAtCurrentIndex(_ element: Element) {
         if currIndex == nil { currIndex = 0 }
         items.insert(element, at: currIndex!)
         fixIndexIntoRange()
     }
 
+    // Given a certain sorted array, this function determines whether to sort it based on a superset of a larger array, or an array simply in a different order.
+    public mutating func reorderItems(to elements: [Element], index: Int) {
+        // If the index is equal to 0, it indicates that the elements passed through as an argument is a larger array, and items is the subset of elements.
+        if index == -1 {
```

Comment:
I'm not a fan of passing an index of -1 to change the behavior of this method. Other than changing how the sorting is done, `index` is only used to update `currIndex`, right?

I also don't quite understand the intended behavior of this method in the two cases. Is this right?
- If `index == -1`, then we resort the internal `items` array, preserving the relative order of items in the `elements` parameter?
- If `index != -1`, then we filter the internal `items` array based on whether each element is in `elements`?

If that is correct, then these two behaviors seem pretty unrelated to each other. And passing `-1` to `index` definitely doesn't make sense to switch between the two to me.

Maybe they should be two separate methods? One to rearrange elements based on another array, and one to filter? Or at least we should come up with a better mechanism to switch between the behavior and pick a better name. You could also add a method to change `currIndex` if needed.

Regardless, let's add some good doc (`///`) comments to whatever methods we add to this repo.

Reply:
You bring up good points, yeah I'll separate the two conditions within the function to be their own independent function. Passing -1 to index was just a way to distinguish two different cases that occurred when the function occurred, but in retrospect it probably would've been better to use enums and make separate functions. I'll fix this and redo the documentation.

Reply:
Cool. I might prefer to provide separate methods and let the caller call the associated method